src/crimson/net/Protocol.cc

```diff
@@ -49,7 +49,6 @@ seastar::future<> Protocol::close()
 
   // unregister_conn() drops a reference, so hold another until completion
   auto cleanup = [conn_ref = conn.shared_from_this(), this] {
-      logger().debug("{} closed!", conn);
```

Comment:
> Serializing a connection appears to access the messenger, which is
problematic as it may not exist when these callbacks fire.

This `cleanup()` is called before `close_ready` becomes available right? And the future returned by `Messenger::close()` is designed to tell messenger users not to free this messenger until this future becomes available, so the messenger **should** be exist inside `cleanup()`.

IMHO it implies a bug from messenger users (not messenger itself) that they are failed to manage messenger lifecycle properly.

Reply:
@cyx1231st #31904

Reply:
Dropped the commit now that #31904 is merged.